dbt/include/athena/macros/materializations/snapshots/snapshot.sql

```diff
@@ -409,15 +154,20 @@
 
   {% if not target_relation_exists %}
 
-      {% set build_sql = build_snapshot_table(strategy, model['compiled_sql'], table_type) %}
+      {% set build_sql = build_snapshot_table(strategy, model['compiled_sql']) %}
       {% set final_sql = create_table_as(False, target_relation, build_sql) %}
 
   {% else %}
 
       {{ adapter.valid_snapshot_target(target_relation) }}
 
+      {% set staging_relation = make_temp_relation(target_relation) %}
```

Comment:
This piece of logic removes the existing staging table in case it wasn't properly removed in the previous run.